### PR TITLE
test(e2e): add quota test

### PR DESCRIPTION
**Pull request by llzcx**

### CODE REVIEW

File: tests/kafkatest/automq/quota_test.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        block_size = 256 * 1024 * 1024
+        server_prop_overrides = [
+            ['broker.quota.enabled', 'true'],
+            ['s3.wal.cache.size', str(log_size)],
+            ['s3.wal.capacity', str(log_size)],
+            ['s3.wal.upload.threshold', str(log_size // 4)],
+            ['s3.block.cache.size', str(block_size)],
+            ['s3.telemetry.metrics.exporter.type', 'otlp'],
+            ['s3.telemetry.exporter.otlp.endpoint', 'http://metrics.hellocorp.site'],
+            ['s3.telemetry.exporter.otlp.protocol', 'grpc'],
```
Comment (ShadowySpirits) [inline tests/kafkatest/automq/quota_test.py:63]: Remove the debug codes.

File: tests/kafkatest/automq/quota_test.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+        self.success = True
+        self.msg = ''
+
+
+    def create_kafka(self, test_context):
+        log_size = 256 * 1024 * 1024
+        block_size = 256 * 1024 * 1024
+        server_prop_overrides = [
+            ['broker.quota.enabled', 'true'],
```
Comment (ShadowySpirits) [inline tests/kafkatest/automq/quota_test.py:56]: Add and verify the static configuration.

- broker.quota.produce.bytes
- broker.quota.fetch.bytes

File: tests/kafkatest/automq/quota_test.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+            ['s3.wal.cache.size', str(log_size)],
+            ['s3.wal.capacity', str(log_size)],
+            ['s3.wal.upload.threshold', str(log_size // 4)],
+            ['s3.block.cache.size', str(block_size)],
+            ['s3.telemetry.metrics.exporter.type', 'otlp'],
+            ['s3.telemetry.exporter.otlp.endpoint', 'http://metrics.hellocorp.site'],
+            ['s3.telemetry.exporter.otlp.protocol', 'grpc'],
+        ]
+        self.kafka = KafkaService(test_context, num_nodes=1, zk=None,
+                                  kafka_heap_opts="-Xmx2048m -Xms2048m",
```
Comment (ShadowySpirits) [inline tests/kafkatest/automq/quota_test.py:66]: Is this a proper option?

File: tests/kafkatest/automq/quota_test.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        self.kafka.start()
+        records = 50000
+        self.logger.info(f'update to {broker_in},{broker_out}')
+        self.test_quota0(records, broker_in, broker_out)
+        # quota reduction
+        broker_in = broker_in // 2
+        broker_out = broker_out // 2
+        self.update_quota_config(broker_in, broker_out)
+        self.logger.info(f'update to {broker_in},{broker_out}')
+        self.test_quota0(records, broker_in, broker_out)
```
[the remark this answers is not in the document]
Comment (llzcx) [reply tests/kafkatest/automq/quota_test.py:115]: when dynamically publishing quota configurations, it is half of the original

**Review by ShadowySpirits — APPROVED** [review]

**Pull request merged**